functions/supabase/seed.sql

```diff
@@ -0,0 +1,101 @@
+create table if not exists users (
+    id text not null primary key,
+    data jsonb not null,
+    fs_updated_time timestamp not null
+);
+alter table users enable row level security;
+
+create table if not exists contracts (
+    id text not null primary key,
+    data jsonb not null,
+    fs_updated_time timestamp not null
+);
+alter table contracts enable row level security;
+
+create table if not exists groups (
+    id text not null primary key,
+    data jsonb not null,
+    fs_updated_time timestamp not null
+);
+alter table groups enable row level security;
+
+create table if not exists txns (
+    id text not null primary key,
+    data jsonb not null,
+    fs_updated_time timestamp not null
+);
+alter table txns enable row level security;
+
+create table if not exists bets (
+    id text not null primary key,
+    data jsonb not null,
+    fs_updated_time timestamp not null
+);
+alter table bets enable row level security;
+
+create table if not exists comments (
+    id text not null primary key,
+    data jsonb not null,
+    fs_updated_time timestamp not null
+);
+alter table comments enable row level security;
+
+create table if not exists incoming_writes (
+  id bigint generated always as identity primary key,
+  event_id text unique null, /* can be null for writes generated by manual import */
+  doc_kind text not null,
+  write_kind text not null,
+  doc_id text not null,
+  data jsonb not null,
+  ts timestamp not null,
+  processed boolean not null default false
+);
+alter table incoming_writes enable row level security;
+
+create or replace function replicate_writes_process()
+  returns trigger
+  language plpgsql
+as
+$$
+declare dest_table text;
+declare replicated_id text;
+begin
+  dest_table = case new.doc_kind
+    when 'txn' then 'txns'
+    when 'user' then 'users'
+    when 'group' then 'groups'
+    when 'contract' then 'contracts'
+    when 'contractBet' then 'bets'
+    when 'contractComments' then 'comments'
+    else null
+  end;
+  if dest_table = null then
+    raise warning 'Invalid document kind.';
+    return new;
+  end if;
+  if new.write_kind = 'create' or new.write_kind = 'update 'then
+    execute format(
+      'insert into %1$I (id, data, fs_updated_time) values (%2$L, %3$L, %4$L)
+       on conflict (id) do update set data = %3$L, fs_updated_time = %4$L
+       where %1$I.fs_updated_time <= %4$L
+       returning id;',
+      dest_table, new.doc_id, new.data, new.ts
+    ) into replicated_id;
```

Comment:
I don't actually know sql. What does this do?

Reply:
This inserts a new row into the replicated table with the ID, data, and timestamp provided. If there is already a row with that ID (i.e. a write for that document is already in the system) then it updates it iff the new update time is later than the update time on that existing row.